### PR TITLE
Restructure contracts

### PR DESCRIPTION
**Pull request by hswick**
Reorganized the contracts in accordance with the spec in whitepaper. Added TaskGiver, Solver, Verifier contracts with test files. Tried to do my best to keep old code for future use, but some of it was deleted for housekeeping sake.

Envisioning the Truebit contract to be the thing that organizes TaskGiver, Solver, and Verifiers. More details need to be discussed.

### CODE REVIEW

**Comment by hswick** [comment]
Pushed some new commits. Put most of Zac's code back in after I realized what broadcasting is about. I added a PostTask event to the Truebit contract which gets emitted when TaskGiver creates a task. 

In the tests when the event is emitted a solver posts a bid. Address of the sender gets sent to the TaskGiver.

**Comment by hswick** [comment]
TaskGiverClient and SolverClient send data to each other via events in the Truebit contract.

File: contracts/Truebit.sol [hunk 1]
```diff
@@ -1,57 +1,70 @@
...
+    PostTask(numTasks, minDeposit);
     numTasks++;
     return numTasks - 1;
   }
 
-  function proccessTask(uint taskIndex){
+  function processTask(uint taskIndex){
     Task t = tasks[taskIndex];
-    require(solved(t));
-    require(!t.proccessed);
```
Comment (mrsmkl) [inline contracts/Truebit.sol]: Why did you remove these two lines?

Comment (hswick) [reply contracts/Truebit.sol]: Must've done that when I was messing around with testing. Added them back.

File: contracts/TaskGiver.sol [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    return taskIndex;
+  }
+
+  //Called by Truebit contract to send the solution back
+  function processTask(uint taskIndex, bytes32 solution) onlyTruebit returns (bool){
+    tasks[taskIndex] = solution;
+    return true;
+  }
+
+  function postBid(address solverAddress, uint taskId) onlyTruebit returns (bool) {
```
Comment (zmitton) [inline contracts/TaskGiver.sol]: I'm a bit confused with this function. I don't  think you enabled `truebit` to call this.

File: scripts/solverClient.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	//Create hash of random bytes and hash of last block
+	this.createRandomHash = function() {
+		this.r = crypto.randomBytes(10).toString('hex');
+		this.hash = web3.sha3(web3.eth.getBlock(web3.eth.blockNumber).hash + this.r);
+		return this.hash;
+	}
+
+	//Check if hash is even or odd
```
Comment (zmitton) [inline scripts/solverClient.js]: from the whitepaper it should check if hash is below a certain threshold, but good for now

**Comment by zmitton** [comment]
im having trouble getting the tests to run. let hold off until we can get to the heart of it

**Comment by hswick** [comment]
Created new PR with new code.
